src/matches.rs

```diff
@@ -143,3 +145,20 @@ fn has_only_ref_pats(arms: &[Arm]) -> bool {
     // look for Some(v) where there's at least one true element
     mapped.map_or(false, |v| v.iter().any(|el| *el))
 }
+
+fn match_template(source: MatchSource, op: &str, expr: &str) -> String {
+    match source {
+        MatchSource::Normal => {
+            format!("match {}{} {{ ...", op, expr)
+        }
+        MatchSource::IfLetDesugar { .. } => {
+            format!("if let ... = {}{} {{", op, expr)
+        }
+        MatchSource::WhileLetDesugar => {
+            format!("while let ... = {}{} {{", op, expr)
+        }
+        MatchSource::ForLoopDesugar => {
+            panic!("for loop desugared to match with &-patterns!")
```

Comment:
This line is `panic!` because it is unreachable. `MatchSource::ForLoopDesugar` is for matching the return value of `Iterator::next` to `Some` and `None`. It is unrelated to & in iterated expressions.